src/asl.ts

```diff
@@ -0,0 +1,1909 @@
+        ? { hasTask: true }
+        : node.kind === "ThrowStmt"
+        ? { hasThrow: true }
+        : {}
+    );
+}
+
+function hasBreak(loop: ForInStmt | ForOfStmt | WhileStmt | DoStmt): boolean {
+  for (const child of loop.children) {
+    if (hasBreak(child)) {
+      return true;
+    }
+  }
+  return false;
+
+  function hasBreak(node: FunctionlessNode): boolean {
+    if (
+      node.kind === "ForInStmt" ||
+      node.kind === "ForOfStmt" ||
+      node.kind === "WhileStmt" ||
+      node.kind === "DoStmt"
+    ) {
+      return false;
+    } else if (node.kind === "BreakStmt") {
+      return true;
+    } else {
+      for (const child of node.children) {
+        if (hasBreak(child)) {
+          return true;
+        }
+      }
+      return false;
+    }
+  }
+}
+
+export namespace ASL {
+  export function toJson(expr: Expr): any {
+    if (expr.kind === "Argument") {
+      return toJson(expr.expr);
+    } else if (expr.kind === "CallExpr") {
+      if (isSlice(expr)) {
+        return sliceToJsonPath(expr);
+      } else if (isFilter(expr)) {
+        return filterToJsonPath(expr);
+      }
+    } else if (expr.kind === "Identifier") {
+      return toJsonPath(expr);
+    } else if (expr.kind === "PropAccessExpr") {
+      return `${toJson(expr.expr)}.${expr.name}`;
+    } else if (
+      expr.kind === "ElementAccessExpr" &&
+      expr.element.kind === "NumberLiteralExpr"
+    ) {
+      return `${toJson(expr.expr)}[${expr.element.value}]`;
+    } else if (expr.kind === "ArrayLiteralExpr") {
+      if (expr.items.find(isVariableReference) !== undefined) {
+        return `States.Array(${expr.items
+          .map((item) => toJsonPath(item))
+          .join(", ")})`;
+      }
+      return expr.items.map((item) => toJson(item));
+    } else if (expr.kind === "ObjectLiteralExpr") {
+      const payload: any = {};
+      for (const prop of expr.properties) {
+        if (prop.kind !== "PropAssignExpr") {
+          throw new Error(
+            `${prop.kind} is not supported in Amazon States Language`
+          );
+        }
+        if (
+          (prop.name.kind === "ComputedPropertyNameExpr" &&
+            prop.name.expr.kind === "StringLiteralExpr") ||
+          prop.name.kind === "Identifier" ||
+          prop.name.kind === "StringLiteralExpr"
+        ) {
+          payload[
+            `${
+              prop.name.kind === "Identifier"
+                ? prop.name.name
+                : prop.name.kind === "StringLiteralExpr"
+                ? prop.name.value
+                : (prop.name.expr as StringLiteralExpr).value
+            }${
+              isLiteralExpr(prop.expr) || isReferenceExpr(prop.expr) ? "" : ".$"
+            }`
+          ] = toJson(prop.expr);
+        } else {
+          throw new Error(
+            `computed name of PropAssignExpr is not supported in Amazon States Language`
+          );
+        }
+      }
+      return payload;
+    } else if (isLiteralExpr(expr)) {
```

Comment:
missing support for unary (`-`, `!`) and binary expressions.

Reply:
Ah dang, you're right.

Reply:
https://github.com/sam-goodwin/functionless/issues/83